Refresh.GameServer/Resources/ResourceHelper.cs

```diff
@@ -8,5 +14,124 @@ public static Stream StreamFromResource(string name)
+            ReadOnlySpan<Vector<byte>> keyVec = key[misalignment..].Cast<byte, Vector<byte>>();
+
+            //Iterate over every item in the data vector
+            for (int i = 0; i < alignedDataVec.Length; i++) 
+                //XOR it with the key
+                alignedDataVec[i] ^= keyVec[i];
+
+            //Get the amount of bytes which are left over
+            int leftoverCount = data.Length - (alignedDataVec.Length * Vector<byte>.Count * sizeof(byte)) - misalignment;
+                
+            //If theres no leftover bytes, return out
+            if (leftoverCount <= 0) return;
+                
+            //Get the start index of the leftover bytes
+            int leftoverOffset = data.Length - leftoverCount;
+            //Slice the data to only have the leftover data
+            Span<byte> leftover = data[leftoverOffset..];
+
+            XorScalar(leftover, key[leftoverOffset..]);
+        }
+    }
+
+    private static void XorScalar(Span<byte> data, ReadOnlySpan<byte> key)
+    {
+        //Iterate over all bytes
+        for (int i = 0; i < data.Length; i++) 
+            //Iterate over all data, and XOR it with the key
+            data[i] ^= key[i];
+    }
+    
+    private static readonly ThreadLocal<Iron> Iron = new(() => new Iron(ArrayPool<byte>.Shared));
+
+    private static void Encrypt(Span<byte> data, ReadOnlySpan<byte> key)
+    {
+        throw new NotImplementedException(); //TODO
+    }
+    
+    /// <summary>
+    /// Decrypt a PSP asset
+    /// </summary>
+    /// <param name="data">The data to decrypt</param>
+    /// <param name="key">The decryption key</param>
+    /// <returns>The decrypted data.</returns>
+    public static byte[] PspDecrypt(Span<byte> data, ReadOnlySpan<byte> key)
+    {
+        //XOR the data
+        Xor(data, key);
+
+        //Get the data buffer
+        Span<byte> buf = data[..^0x19];
+        //Get the info of the file
+        Span<byte> info = data[^0x19..];
+
+        uint len = BinaryPrimitives.ReadUInt32LittleEndian(info[..4]);
+
+        ReadOnlySpan<byte> signature = info[^0x10..];
+
+        //If the hash in the file does not match the data contained, the file is likely corrupt.
+        if (!signature.SequenceEqual(MD5.HashData(data[..^0x10]))) throw new InvalidDataException("Encrypted PSP asset hash does not match signature, likely corrupt");
+        
+        //If the data is not compressed, just return a copy of the raw buffer
+        if (info[0x4] != 1)
+        {
+            Xor(data, key);
+            return buf.ToArray();
+        }
+
+        //Decompress the data
+        using IronCompressResult decompressed = Iron.Value!.Decompress(Codec.LZO, buf, (int)len);
+
+        Xor(data, key);
+
+        //Return a copy of the decompressed data
+        return decompressed.AsSpan().ToArray();
+    }
+    
+    public static byte[] XorAlloc(ReadOnlySpan<byte> data, ReadOnlySpan<byte> key)
```

Comment:
Is this function used anywhere?

Reply:
Not atm, its a leftover, its unneeded now, i'll remove it